curvefs/src/client/warmup/warmup_manager.cpp

```diff
@@ -93,13 +96,74 @@ bool WarmupManagerS3Impl::AddWarmupFile(fuse_ino_t key, const std::string& path,
         return false;
     }
     // add warmup Progress
-    if (AddWarmupProcess(key, type)) {
+    if (AddWarmupProcess(key, path, type)) {
         LOG(INFO) << "add warmup single task:" << key;
         FetchDentryEnqueue(key, path);
     }
     return true;
 }
 
+bool WarmupManagerS3Impl::CancelWarmupFileOrFilelist(fuse_ino_t key) {
+    if (!mounted_.load(std::memory_order_acquire)) {
+        LOG(ERROR) << "not mounted";
+        return false;
+    }
+    if (CancelWarmupProcess(key)) {
+        VLOG(9) << "cancel warmup list task: " << key;
+        WriteLockGuard lock(warmupFilelistDequeMutex_);
+        auto iter = FindWarmupFilelistByKeyLocked(key);
+        if (iter != warmupFilelistDeque_.end()) {
+            std::shared_ptr<InodeWrapper> inodeWrapper;
+            CURVEFS_ERROR ret = inodeManager_->GetInode(key, inodeWrapper);
+            if (ret != CURVEFS_ERROR::OK) {
+                LOG(ERROR) << "inodeManager get inode fail, ret = " << ret
+                           << ", inodeid = " << key;
+                return false;
+            }
+            uint64_t len = inodeWrapper->GetLength();
```

Comment:
What is this section used for?

Reply:
To ensure the target inode exists in the file system